### PR TITLE
Remove deprecated API added to the compatibility script

### PR DESCRIPTION
**Pull request by karimJWP**
### This PR will...
remove all instances of deprecated API methods added to compatibility script.

### Why is this Pull Request needed?
To exclude deprecated methods from jw8

### Are there any points in the code the reviewer needs to double check?
n/a -> callbacks are removed in https://github.com/jwplayer/jwplayer/pull/2157/
### Are there any Pull Requests open in other repos which need to be merged with this?
https://github.com/jwplayer/jwplayer-ads-vast/pull/210

#### Addresses Issue(s):

JW8-110

### CODE REVIEW

File: test/data/api-methods.js [hunk 1]
```diff
@@ -40,7 +36,6 @@ define({
     getPosition: null,
     getProvider: null,
     getQualityLevels: null,
-    getRenderingMode: null,
     getSafeRegion: null,
     getState: null,
```
Comment (robwalch) [inline test/data/api-methods.js]: Now that we have a list of deprecated API methods in the test/data, can you add the ones you've remove here to that file?

Comment (karimJWP) [inline test/data/api-methods.js]: @robwalch all set and no unit test issues

**Comment by karimJWP** [comment]
test this please

**Comment by robwalch** [comment]
The build failure is because a test dependency in commercial against this project I'm in the process of fixing.

**Comment by robwalch** [comment]
test this please

**Pull request merged**